### PR TITLE
Update IDP signin status API explainer

### PR DESCRIPTION
**Pull request by cbiesinger**
Per discussion at TPAC.

### CODE REVIEW

**Comment by cbiesinger** [comment]
@samuelgoto @martinthomson @bvandersloot-mozilla -- could you sanitycheck that this represents what we discussed?

**Comment by cbiesinger** [comment]
Also should I rename this to "IDP Login Status API" now, since we changed signin to login in the API?

File: proposals/idp-sign-in-status-api.md [hunk 1]
```diff
@@ -51,23 +51,30 @@ The signout-all header should only be sent when no accounts remain signed in
...
 ```idl
 
-dictionary SigninStatusOptions {
-  boolean idp = false;
+enum LoginStatus {
+  "logged-in",
+  "logged-out",
+};
+
+interface NavigatorLogin {
```
Comment (npm1) [inline proposals/idp-sign-in-status-api.md:66]: Is this the right name? I see `NavigatorLogin` but elsewhere I saw `Login`

**Review by npm1 — COMMENTED** [review]
If we want to keep this updated, I think there are some more changes needed:

* A few places have "signed-in" or "signed-out" where it should use "logged-in" or "logged-out"
* Should the state naming also change?
* The alternatives considered seciton also needs to be updated. In particular, it still mentions "action=signin".

File: proposals/idp-sign-in-status-api.md (hunk 1 above)
Comment (cbiesinger) [reply proposals/idp-sign-in-status-api.md:66]: As discussed, keeping NavigatorLogin to reduce the risk of naming conflicts with global objects.

I've made the other requested changes

File: proposals/idp-sign-in-status-api.md [hunk 2]
```diff
@@ -47,9 +47,9 @@ Set-Login: logged-out
 These headers can be sent on the toplevel load as well as subresources such as
 XMLHttpRequest (this is necessary for at least one IDP).
 
-The signout-all header should only be sent when no accounts remain signed in
-to the IDP, i.e. when this action has signed out all accounts or if this
-was the last/only account getting signed out.
+The signout header should only be sent when no accounts remain logged in
+to the IDP, i.e. when this action has logged out all accounts or if this
```
Comment (TallTed) [inline proposals/idp-sign-in-status-api.md:51]:
```suggestion
to the IDP, i.e., when this action has logged out all accounts or if this
```

File: proposals/idp-sign-in-status-api.md [hunk 3]
```diff
@@ -75,7 +75,7 @@ partial interface Navigator {
 Alternatively, an IdP can call the IdP Sign-in Status API via JS calls through
 the static functions `navigator.login.setStatus("logged-in")` and
 `navigator.login.setStatus("logged-out")`. These are to be called from the IDP's
-origin, and mark the current origin as signed in or signed out.
+origin, and mark the current origin as logged in or logged out.
```
Comment (TallTed) [inline proposals/idp-sign-in-status-api.md:78]:
```suggestion
origin, and mark the current origin as logged-in or logged-out.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: proposals/idp-sign-in-status-api.md [hunk 4]
```diff
@@ -154,31 +162,28 @@ See [https://github.com/fedidcg/FedCM/blob/main/meetings/2022/FedCM_%20Options%2
...
-sent when all accounts from this IDP are signed out.
-
 We could instead or in addition have allowed notifying the user agent of
-individual accounts being signed in/out, such as:
+individual accounts being logged in/out, such as:
 
 ```
-SignIn-Status: action=signin; count=2; type=idp
-SignIn-Status: action=signout; new-count=1; type=idp
+SignIn-Status: signin; count=2; type=idp
```
Comment (samuelgoto) [inline proposals/idp-sign-in-status-api.md:169]:
```suggestion
Set-Login: logged-in;  count=2; type=idp
```

File: proposals/idp-sign-in-status-api.md [hunk 5]
```diff
@@ -154,31 +162,28 @@ See [https://github.com/fedidcg/FedCM/blob/main/meetings/2022/FedCM_%20Options%2
...
+SignIn-Status: signout; new-count=1; type=idp
```
Comment (samuelgoto) [inline proposals/idp-sign-in-status-api.md:170]:
```suggestion
Set-Login: logged-out; new-count=1; type=idp
```

File: proposals/idp-sign-in-status-api.md [hunk 6]
```diff
@@ -154,31 +162,28 @@ See [https://github.com/fedidcg/FedCM/blob/main/meetings/2022/FedCM_%20Options%2
...
 ```
 
 Or
 
 ```
-SignIn-Status: action=signin; accountid=foo@bar.com; type=idp
-SignIn-Status: action=signout; accountid=foo@bar.com; type=idp
+SignIn-Status: signin; accountid=foo@bar.com; type=idp
```
Comment (samuelgoto) [inline proposals/idp-sign-in-status-api.md:176]:
```suggestion
Set-Login: logged-in; accountid=foo@bar.com; type=idp

```

File: proposals/idp-sign-in-status-api.md [hunk 7]
```diff
@@ -154,31 +162,28 @@ See [https://github.com/fedidcg/FedCM/blob/main/meetings/2022/FedCM_%20Options%2
...
+SignIn-Status: signout; accountid=foo@bar.com; type=idp
```
Comment (samuelgoto) [inline proposals/idp-sign-in-status-api.md:177]:
```suggestion
Set-Login: logged-out;  accountid=foo@bar.com; type=idp
```

**Comment by samuelgoto** [comment]
Just wondering, since there is a lot of duplication between the two explainer, why wouldn't we just point people to this explainer instead [1] for the specifics, and this explainer can go over how FedCM intends to use the signal ?

https://github.com/fedidcg/login-status

File: proposals/idp-sign-in-status-api.md [hunk 8]
```diff
@@ -51,8 +51,9 @@ The signout header should only be sent when no accounts remain logged in
 to the IDP, i.e., when this action has logged out all accounts or if this
 was the last/only account getting logged out.
 
-We envision this to be a semicolon-separated list of tokens or key/value pairs
-so that this can be expanded in the future.
+This will be parsed as a [structured field](https://fetch.spec.whatwg.org/#concept-header-list-get-structured-header)
+with type `item`. For now we only use bare item part but this lets us extend
+the header to support parameters in the future.
```
Comment (TallTed) [inline proposals/idp-sign-in-status-api.md:56]:
```suggestion
with type `item`. For now, we only use bare item part, but this lets us extend
the header to support parameters in the future.
```

Comment (cbiesinger) [reply proposals/idp-sign-in-status-api.md]: Done

File: proposals/idp-sign-in-status-api.md (hunk 4 above)
Comment (cbiesinger) [reply proposals/idp-sign-in-status-api.md]: Done

File: proposals/idp-sign-in-status-api.md (hunk 6 above)
Comment (cbiesinger) [reply proposals/idp-sign-in-status-api.md]: Done

File: proposals/idp-sign-in-status-api.md (hunk 7 above)
Comment (cbiesinger) [reply proposals/idp-sign-in-status-api.md]: Done

**Review by cbiesinger — COMMENTED** [review]
I addressed the comments.

I added a link to the login status explainer but wanted to keep the two sections in this document so readers don't need to read two documents to understand the proposal.

**Pull request merged**